Editor/SettingsAnalysis/PlayerSettingsAnalyzer.cs

```diff
@@ -103,5 +150,23 @@ internal static void FixSpeakerMode()
 
             AudioSettings.Reset(audioConfiguration);
         }
+
+        internal static bool CheckIL2CPPCompilerConfiguration(Il2CppCompilerConfiguration compilerConfiguration)
+        {
+            var buildTargetGroup = EditorUserBuildSettings.selectedBuildTargetGroup;
```

Comment:
the selected build target is not necessarily the platform we are analyzing. You can get that via `projectAuditorParams`